### PR TITLE
fix etcd watcher goroutine leak test failure

### PR DESCRIPTION
**Pull request by lysu**
our testcase sometime failured by etcd watcher goroutine leak which is caused by miss close etcdclient/domain/storage in unit-test.

this PR fixes some protenial point(maybe not full) .

```
case leak status ERROR EXITED – ["2018/08/15 19:22:09 main.go:103: \u001b[0;31m[fatal] goroutine seems to leak:\ngoroutine 849 [select, 20 minutes]:\ngithub.com/pingcap/tidb/vendor/google.golang.org/grpc/transport.(*recvBufferReader).read(0xc4211ed590, 0xc42106f390, 0x5, 0x5, 0x1, 0xc4209fa9d0, 0x1)\n\t/home/jenkins/workspace/tidb_ghpr_build/go/src/github.com/pingcap/tidb/vendor/google.golang.org/grpc/transport/transport.go:142 +0x1e9\ngithub.com/pingcap/tidb/vendor/google.golang.org/grpc/transport.(*recvBufferReader).Read(0xc4211ed590, 0xc42106f390, 0x5, 0x5, 0xc42b826f1f, 0x34, 0x34)\n\t/home/jenkins/workspace/tidb_ghpr_build/go/src/github.com/pingcap/tidb/vendor/google.golang.org/grpc/transport/transport.go:131 +0x67\ngithub.com/pingcap/tidb/vendor/google.golang.org/grpc/transport.(*transportReader).Read(0xc42072e720, 0xc42106f390, 0x5, 0x5, 0xc4209fabc4, 0xc4209faa68, 0x982c58)\n\t/home/jenkins/workspace/tidb_ghpr_build/go/src/github.com/pingcap/tidb/vendor/google.golang.org/grpc/transport/transport.go:369 +0x55\nio.ReadAtLeast(0x1430fe0, 0xc42072e720, 0xc42106f390, 0x5, 0x5, 0x5, 0xc420582b40, 0xc420d94b40, 0xc400000005)\n\t/usr/local/go/src/io/io.go:309 +0x86\nio.ReadFull(0x1430fe0, 0xc42072e720, 0xc42106f390, 0x5, 0x5, 0x0, 0xc4203e7380, 0xc4203e72c0)\n\t/usr/local/go/src/io/io.go:327 +0x58\ngithub.com/pingcap/tidb/vendor/google.golang.org/grpc/transport.(*Stream).Read(0xc420d94b40, 0xc42106f390, 0x5, 0x5, 0x3, 0x1435d00, 0xc420223f80)\n\t/home/jenkins/workspace/tidb_ghpr_build/go/src/github.com/pingcap/tidb/vendor/google.golang.org/grpc/transport/transport.go:353 +0xbf\ngithub.com/pingcap/tidb/vendor/google.golang.org/grpc.(*parser).recvMsg(0xc42106f380, 0x7fffffff, 0x456827, 0xc4209faed0, 0x42cb3b, 0xc4209fac30, 0x455d90, 0xc420483b00)\n\t/home/jenkins/workspace/tidb_ghpr_build/go/src/github.com/pingcap/tidb/vendor/google.golang.org/grpc/rpc_util.go:452 +0x65\ngithub.com/pingcap/tidb/vendor/google.golang.org/grpc.recv(0xc42106f380, 0x7fd7293885a8, 0x1e82500, 0xc420d94b40, 0x0, 0x0, 0x12b9360, 0xc4451e9860, 0x7fffffff, 0x0, ...)\n\t/home/jenkins/workspace/tidb_ghpr_build/go/src/github.com/pingcap/tidb/vendor/google.golang.org/grpc/rpc_util.go:561 +0x4d\ngithub.com/pingcap/tidb/vendor/google.golang.org/grpc.(*csAttempt).recvMsg(0xc420aabc70, 0x12b9360, 0xc4451e9860, 0x0, 0x0)\n\t/home/jenkins/workspace/tidb_ghpr_build/go/src/github.com/pingcap/tidb/vendor/google.golang.org/grpc/stream.go:529 +0x132\ngithub.com/pingcap/tidb/vendor/google.golang.org/grpc.(*clientStream).RecvMsg(0xc420f84700, 0x12b9360, 0xc4451e9860, 0x50, 0x126d3a0)\n\t/home/jenkins/workspace/tidb_ghpr_build/go/src/github.com/pingcap/tidb/vendor/google.golang.org/grpc/stream.go:395 +0x43\<http://ngithub.com/pingcap/tidb/vendor/github.com/grpc-ecosystem/go-grpc-prometheus.(*monitoredClientStream).RecvMsg(0

xc42106f3a0|ngithub.com/pingcap/tidb/vendor/github.com/grpc-ecosystem/go-grpc-prometheus.(*monitoredClientStream).RecvMsg(0xc42106f3a0>, 0x12b9360, 0xc4451e9860, 0x0, 0x0)\n\t/home/jenkins/workspace/tidb_ghpr_build/go/src/github.com/pingcap/tidb/vendor/github.com/grpc-ecosystem/go-grpc-prometheus/client.go:63 +0x4b\ngithub.com/pingcap/tidb/vendor/github.com/coreos/etcd/etcdserver/etcdserverpb.(*watchWatchClient).Recv(0xc420c31050, 0x0, 0x0, 0x0)\n\t/home/jenkins/workspace/tidb_ghpr_build/go/src/github.com/pingcap/tidb/vendor/github.com/coreos/etcd/etcdserver/etcdserverpb/rpc.pb.go:3217 +0x62\ngithub.com/pingcap/tidb/vendor/github.com/coreos/etcd/clientv3.(*watchGrpcStream).serveWatchClient(0xc420aabad0, 0x144d120, 0xc420c31050)\n\t/home/jenkins/workspace/tidb_ghpr_build/go/src/github.com/pingcap/tidb/vendor/github.com/coreos/etcd/clientv3/watch.go:565 +0x4e\ncreated by github.com/pingcap/tidb/vendor/github.com/coreos/etcd/clientv3.

(*watchGrpcStream).newWatchClient\n\t/home/jenkins/workspace/tidb_ghpr_build/go/src/github.com/pingcap/tidb/vendor/github.com/coreos/etcd/clientv3/watch.go:720 +0x58f--------10.144.7.234 2018-08-15 19:22:09.376126917 +0800 CST m=+33542.448973149 \u001b[0m\n2018/08/15 19:22:09 daemon.go:225: \u001b[0;33m[warning] daemon leak exit with error exit status 255\u001b[0m\n2018/08/15 19:26:49 process_handler.go:97: \u001b[0;36m[debug] state ERROR EXITED\u001b[0m\n2018/08/15 19:26:53 process_handler.go:97: \u001b[0;36m[debug] state ERROR EXITED\u001b[0m\n"]
```

### What is changed and how it works?

- close etcdclient in test
- close domain/storage in testcase
- use right method name for test TearDown

Tests 

 - Unit test
 - Integration test

Code changes

 - test case change

Side effects

 - No

Related changes

 - No related changes

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/7408)

### CODE REVIEW

File: session/bench_test.go [hunk 1]
```diff
@@ -98,246 +100,331 @@ func readResult(ctx context.Context, rs ast.RecordSet, count int) {
 
 func BenchmarkBasic(b *testing.B) {
 	ctx := context.Background()
-	b.StopTimer()
-	se := prepareBenchSession()
-	b.StartTimer()
+	se, do, st := prepareBenchSession()
+	defer func() {
+		do.Close()
```
Comment (tiancaiamao) [inline session/bench_test.go]: This order:
```
se.Close()
st.Close()
do.Close()
```

File: session/bench_test.go [hunk 2]
```diff
@@ -98,246 +100,331 @@ func readResult(ctx context.Context, rs ast.RecordSet, count int) {
...
+	b.StopTimer()
 }
 
 func BenchmarkTableScan(b *testing.B) {
 	ctx := context.Background()
-	b.StopTimer()
-	se := prepareBenchSession()
+	se, do, st := prepareBenchSession()
+	defer func() {
+		do.Close()
```
Comment (tiancaiamao) [inline session/bench_test.go]: ditto

File: session/bench_test.go [hunk 3]
```diff
@@ -98,246 +100,331 @@ func readResult(ctx context.Context, rs ast.RecordSet, count int) {
...
+	b.StopTimer()
 }
 
 func BenchmarkExplainTableScan(b *testing.B) {
 	ctx := context.Background()
-	b.StopTimer()
-	se := prepareBenchSession()
+	se, do, st := prepareBenchSession()
+	defer func() {
+		do.Close()
```
Comment (tiancaiamao) [inline session/bench_test.go]: ditto

File: store/tikv/lock_resolver.go [hunk 4]
```diff
@@ -52,37 +49,6 @@ func newLockResolver(store Storage) *LockResolver {
 	return r
 }
 
-// NewLockResolver creates a LockResolver.
-// It is exported for other services to use. For instance, binlog service needs
-// to determine a transaction's commit state.
-func NewLockResolver(etcdAddrs []string, security config.Security) (*LockResolver, error) {
```
Comment (tiancaiamao) [inline store/tikv/lock_resolver.go]: This is removed?

File: privilege/privileges/cache_test.go [hunk 5]
```diff
@@ -190,8 +190,10 @@ func (s *testCacheSuite) TestAbnormalMySQLTable(c *C) {
 	session.SetSchemaLease(0)
```
[the remark this answers is not in the document]
Comment (lysu) [reply privilege/privileges/cache_test.go]: done~

File: store/tikv/lock_resolver.go (hunk 4 above)
Comment (lysu) [reply store/tikv/lock_resolver.go]: yes~I found the method is unused code

Comment (tiancaiamao) [reply store/tikv/lock_resolver.go]: Binlog use it!

Comment (tiancaiamao) [reply store/tikv/lock_resolver.go]: Maybe you could update the comment, add "DO NOT REMOVE IT"

Comment (lysu) [reply store/tikv/lock_resolver.go]: done

**Comment by coocood** [comment]
LGTM

**Pull request merged**